mtp_cashbook/apps/mtp_auth/forms.py

```diff
@@ -0,0 +1,53 @@
+from django import forms
+from django.contrib.auth import authenticate
+
+from django.utils.translation import ugettext_lazy as _
+
+from .exceptions import ConnectionError
+
+
+class AuthenticationForm(forms.Form):
+    """
+    Authentication form used for authenticating users during the login process.
+    """
+    username = forms.CharField(label=_("Username"), max_length=254)
+    password = forms.CharField(label=_("Password"), widget=forms.PasswordInput)
+
+    error_messages = {
+        'invalid_login': _("Please enter a correct username and password. "
+                           "Note that both fields may be case-sensitive."),
+        'connection_error': _("The API Server seems down, please try again later."),
+    }
+
+    def __init__(self, request=None, *args, **kwargs):
+        self.request = request
+        self.user_cache = None
+        super(AuthenticationForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
```

Comment:
This doesn't feel like `clean` to me: it seems like this it's getting the cleaned inputs and then doing the actual login… Oh, wait, is this a holdover of this being a subclass of `django.forms.Form`? If so, it might be nice to break this out more clearly into a `login` method that the `clean` method calls, perhaps?